### PR TITLE
Allow to override output format and action fixes

### PR DESCRIPTION
**Pull request by msugakov**
This includes and supersedes https://github.com/stackrox/kube-linter-action/pull/6

Changes:
* Added action parameter for output format.
* Refreshed parameters info in `README.md`.
* Fixed release info caching and api throttling issue.
* Fixed the way kube-linter release version is looked up.
* Extended self-tests.
* Other minor fixes.

Testing done: relying on CI.

### CODE REVIEW

File: .github/workflows/self-test.yml [hunk 1]
```diff
@@ -10,56 +10,66 @@ on:
...
 
 jobs:
-  test-scan-linux:
-    runs-on: ubuntu-latest
+  test-scan:
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest, macos-latest ]
+        format: [ plain, json, sarif ]
+        version: [ latest, 0.2.3 ]
```
Comment (viswajithiii) [inline .github/workflows/self-test.yml:18]: Nice

File: action.yml [hunk 2]
```diff
@@ -10,26 +10,38 @@ inputs:
...
           *)       OS=linux ;;
         esac
-        RELEASE_INFO=$(curl --silent --show-error --fail https://api.github.com/repos/stackrox/kube-linter/releases/${{ inputs.version }})
+        RELEASE_URL='https://api.github.com/repos/stackrox/kube-linter/releases/latest'
+        if [ "${{ inputs.version }}" != "latest" ]; then
+          RELEASE_URL='https://api.github.com/repos/stackrox/kube-linter/releases/tags/${{ inputs.version }}'
+        fi
+        RELEASE_INFO=$(curl --silent --show-error --fail \
+          --header 'authorization: Bearer ${{ github.token }}' \
+          --header 'Cache-Control: no-cache, must-revalidate' \
```
Comment (viswajithiii) [inline action.yml:43]: Maybe add a comment about why this is necessary?

File: action.yml [hunk 3]
```diff
@@ -40,16 +52,13 @@ runs:
...
-        set +e
         if [ -z ${{ inputs.config }} ]; then
           CONFIG=""
         else
           CONFIG="--config ${{ inputs.config }}"
         fi
-        ./kube-linter $CONFIG lint ${{ inputs.directory }} 2>&1 | tee -a ${{ inputs.output-file }}
-        exit ${PIPESTATUS[0]}
-      shell: bash
+        ./kube-linter $CONFIG lint ${{ inputs.directory }} --format ${{ inputs.format }} | tee ${{ inputs.output-file }}
```
Comment (viswajithiii) [inline action.yml:64]: Why did you remove the `-a`?

Comment (viswajithiii) [reply action.yml:64]: How come the `set +e` with `exit ${PIPESTATUS[0]}` was necessary before but isn't now? Or was it just never necessary before?

File: action.yml [hunk 4]
```diff
@@ -10,26 +10,38 @@ inputs:
...
           macOS)   OS=darwin ;;
           Windows) OS=windows ;;
           *)       OS=linux ;;
         esac
-        RELEASE_INFO=$(curl --silent --show-error --fail https://api.github.com/repos/stackrox/kube-linter/releases/${{ inputs.version }})
+        RELEASE_URL='https://api.github.com/repos/stackrox/kube-linter/releases/latest'
+        if [ "${{ inputs.version }}" != "latest" ]; then
+          RELEASE_URL='https://api.github.com/repos/stackrox/kube-linter/releases/tags/${{ inputs.version }}'
+        fi
+        RELEASE_INFO=$(curl --silent --show-error --fail \
```
Comment (viswajithiii) [inline action.yml:41]:
```suggestion
        RELEASE_INFO="$(curl --silent --show-error --fail \
```

File: action.yml [hunk 5]
```diff
@@ -10,26 +10,38 @@ inputs:
...
+          --header 'authorization: Bearer ${{ github.token }}' \
+          --header 'Cache-Control: no-cache, must-revalidate' \
+          "${RELEASE_URL}")
```
Comment (viswajithiii) [inline action.yml:44]:
```suggestion
          "${RELEASE_URL}")"
```

File: action.yml (hunk 4 above)
Comment (msugakov) [reply action.yml]: Done.

File: action.yml (hunk 5 above)
Comment (msugakov) [reply action.yml]: Done.

File: action.yml (hunk 2 above)
Comment (msugakov) [reply action.yml:47]: Added.

File: action.yml (hunk 3 above)
Comment (msugakov) [reply action.yml]: > Why did you remove the `-a`?

Appending log file only makes sense when the output format is `plain`. However, users can run the action multiple times with different formats but using the default output file name. In this case `plain` output would get appended to `json` or `sarif` messing up the output file.

> How come the `set +e` with `exit ${PIPESTATUS[0]}` was necessary before but isn't now?

I think the idea was to return exit code from `kube-linter` only and ignore exit code from `tee`.
GitHub starts bash commands with `set -eo pipefail` already by default so the script previously had `set +e` to allow capturing `kube-linter`-s exit code. Then it would return only this exit code by using `${PIPESTATUS[0]}`.
First, I moved `set +e` closer to where `kube-linter` was started. Then I played with self-test and saw that a case where `output-file` is in not-existing directory does not fail. See https://github.com/stackrox/kube-linter-action/runs/3743081013#step:3:45
Therefore, the right thing is to let any command of the `kube-linter | tee` pipeline fail the action. This is simply achieved by relying on GitHub's defaults.

**Review by viswajithiii — APPROVED** [review]
